docs/iris/src/userguide/real_and_lazy_data.rst

```diff
@@ -1,246 +1,211 @@
-enables the user to load and manipulate datasets that you would otherwise not
-be able to fit into memory.
+In Iris, we use the term **real data** to describe data arrays that are loaded
+into memory. Real data is typically provided as a
+`NumPy array <https://docs.scipy.org/doc/numpy/reference/generated/numpy.array.html>`_,
+which has a shape and data type that are used to describe the array's data points.
+Each data point takes up a small amount of memory, which means large NumPy arrays can
+take up a large amount of memory.
 
-However, in order to execute certain operations (such as calculations on actual
-data values) the real data must be realized.  Using Dask, the operation will
-be deferred until you request the result, at which point it will be executed
-using Dask's parallel processing schedulers.  The combination of these two
-behaviours can offer a significant performance boost.
+Conversely, we use the term **lazy data** to describe data that is not loaded into memory.
+(This is sometimes also referred to as **deferred data**.)
+In Iris, lazy data is provided as a
+`dask array <http://dask.pydata.org/en/latest/array-overview.html>`_.
+A dask array also has a shape and data type
+but typically the dask array's data points are not loaded into memory.
+Instead the data points are stored on disk and only loaded into memory in
+small chunks when absolutely necessary.
 
-You can check whether the data array on your cube is lazy using the Iris
-function 'has_lazy_data'.  For example:
+The primary advantage of using lazy data is that it enables the loading and manipulating
+of datasets that would otherwise not fit into memory.
 
-.. doctest::
+You can check whether a cube has real data or lazy data by using the method
+:meth:`~iris.cube.Cube.has_lazy_data`. For example::
 
-    >>> import iris
     >>> cube = iris.load_cube(filename, 'air_temp.pp')
     >>> cube.has_lazy_data()
     True
     >>> cube.data
     >>> cube.has_lazy_data()
     False
 
-When does my Data Become Real?
+When does my data become real?
 ------------------------------
 
-If the data on your cube is in its lazy state, it will only become real if you
-'touch' the data.  This means any way of directly accessing the data, such as
-assigning it to a variable or simply using 'cube.data' as in the example above.
-
-Any action which requires the use of actual data values (such as cube maths)
-will also cause the data to be loaded into memory, although data realization
-is always deferred until the result is requested:
-
-.. doctest::
+Most operations on data arrays can be run equivalently on both real and lazy data.
+If the data array is real then the operation will be run on the data array
+immediately with the results becoming available as soon as processing is completed.
+If the data array is lazy then the operation will be deferred until you request
+the result (such as when you call ``cube.data``). In this case the data array will
+remain lazy::
```

Comment:
This is a little confusing.  It seems like you are saying that in the case that you request the result (by calling 'cube.data') the array will remain lazy.  Maybe just needs a little clarification of which case is 'this case'.